tests/test_selenium.py

```diff
@@ -53,6 +75,350 @@ def test_signup(selenium, live_server, client):
+        owner=user.profile)
+    buck.save()
+    InvestmentStockConfiguration(quantity=1, stock=stock, bucket=buck, start="2016-03-03").save()
+    assert user.profile.owned_bucket.count() == 1
+    user.profile.default_acc().trade_bucket(buck, .01)
+    selenium.get('%s%s' % (live_server, '/login'))
+    selenium.add_cookie({
+        'name': 'sessionid',
+        'value': cookie.value,
+        'secure': False,
+        'path': '/',
+    })
+    selenium.get('%s%s' % (live_server, '/home'))
+    selenium.implicitly_wait(10)
+    selenium.find_element_by_xpath("//button[contains(.,'Invest')]").click()
+    slider = selenium.find_element_by_class_name("rc-slider-handle")
+    actions = ActionChains(selenium)
+    actions.click_and_hold(slider)
+    actions.move_by_offset(-100, 0)
+    actions.release(slider)
+    actions.perform()
+    save_composition = selenium.find_element_by_id("save")
+    save_composition.click()
+    assert user.profile.trading_accounts.get(account_name='default').buckettrades.count() == 2
+
+
+@mock_plaid_balance_for_trading
+@mock_plaid_accounts
+@mock_plaid_transactions
+@pytest.mark.django_db(transaction=True)
+def test_change_bucket_composition(selenium, live_server, client):
+    """
+    Test selling a bucket
+    """
+    user = User.objects.create_user('temporary', 'temporary@gmail.com', 'temporary')
+    user.save()
+    user.userbank.create(
+        item_id='dummy1', access_token='dummy2',
+        institution_name='dummy3', current_balance_field=0,
+        account_name_field="dummy4", income_field=0,
+        expenditure_field=0
+    )
+    stock = Stock(
+        name="Name1", ticker="poooooop"
+    )
+    stock.save()
+    stock.daily_quote.create(
+        value=1000, date="2016-03-03"
+    )
+    client.login(username='temporary', password='temporary')
+    cookie = client.cookies['sessionid']
+    buck = InvestmentBucket.create_new_bucket(
+        name="IAMATESTBUCKET",
+        public=True,
+        owner=user.profile)
+    buck.save()
+    InvestmentStockConfiguration(quantity=1, stock=stock, bucket=buck, start="2016-03-03").save()
+    assert user.profile.owned_bucket.count() == 1
+    user.profile.default_acc().trade_bucket(buck, .01)
+    selenium.get('%s%s' % (live_server, '/login'))
+    selenium.add_cookie({
+        'name': 'sessionid',
+        'value': cookie.value,
+        'secure': False,
+        'path': '/',
+    })
+    selenium.get('%s%s' % (live_server, '/home'))
+    selenium.implicitly_wait(10)
+    edit_button = selenium.find_element_by_id("edit-comp")
+    edit_button.click()
+    slider = selenium.find_element_by_class_name("rc-slider-handle-2")
+    actions = ActionChains(selenium)
+    actions.click_and_hold(slider)
+    actions.move_by_offset(-100, 0)
+    actions.release(slider)
+    actions.perform()
+    save_composition = selenium.find_element_by_xpath("//button[contains(.,'Save')]")
+    save_composition.click()
+    time.sleep(1)
+    assert buck.stocks.count() == 2
```

Comment:
What does this test do?

Reply:
checks to see that a modification creates another stock config

Reply:
Got it, the user had some bucket and sells (part of) it. Could you also assert that the amount of the bucket he holds is less than currently?

Reply:
Nah this isn't a sell, its a bucket modification, like changing the amount of AAPL in a bucket from 10 to 5. Sell is in the test_invest_sell_bucket, which moves it to 0.